diff -Naur blfs-bootscripts-20041227.orig/blfs/init.d/random blfs-bootscripts-20041227/blfs/init.d/random
--- blfs-bootscripts-20041227.orig/blfs/init.d/random	2004-10-02 17:57:32.000000000 +0000
+++ blfs-bootscripts-20041227/blfs/init.d/random	2005-01-06 07:50:48.583838783 +0000
@@ -17,6 +17,9 @@
 		if [ -f /var/tmp/random-seed ]; then
 			/bin/cat /var/tmp/random-seed >/dev/urandom
 		fi
+		if [ -c /dev/frandom ]; then
+			/bin/dd if=/dev/frandom of=/dev/null count=1
+		fi
 		/bin/dd if=/dev/urandom of=/var/tmp/random-seed \
 			count=1 &>/dev/null
 		evaluate_retval
