diff -Naur hexchat-2.14.3.orig/src/common/servlist.c hexchat-2.14.3/src/common/servlist.c
--- hexchat-2.14.3.orig/src/common/servlist.c	2019-12-21 07:43:47.699069700 +0100
+++ hexchat-2.14.3/src/common/servlist.c	2021-05-21 22:25:53.212588361 +0200
@@ -237,6 +237,9 @@
 	/* Self signed */
 	{0,			"irc.librairc.net"},
 
+	{"Libera Chat", 0, 0, 0, LOGIN_SASL, 0, TRUE},
+	{0,			"irc.libera.chat"},
+
 #ifdef USE_OPENSSL
 	{"LinkNet",	0},
 	{0,			"irc.link-net.org/+7000"},
